ServerDevcommands/Commands/Dmg.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ServerDevcommands;
+public class DmgCommand
+{
+    public DmgCommand()
+    {
+        Helper.Command("dmg", "[target,,value] - Default value: 5. (Negative values heal the character).", (args) =>
+        {
+            if (args == null || args.Length < 2)
+            {
+                Console.instance.Print("Error: Insufficient arguments.");
+                return;
+            }
+
+            if (Player.m_localPlayer == null)
+            {
+                Console.instance.Print("Error: Local player not found.");
+                return;
+            }
+
+            string commandArgs = args.FullLine.Substring(4);
+            string[] parts = commandArgs.Split(new string[] { ",," }, StringSplitOptions.None);
+
+            string argName = parts.Length >= 1 && !string.IsNullOrWhiteSpace(parts[0]) ? parts[0] : "all";
+            float argDmg = parts.Length >= 2 && float.TryParse(parts[1].Trim(), out float parsedDmg) ? parsedDmg : 5f;
+
+            float absDmg = Math.Abs(argDmg);
+            string action = argDmg >= 0 ? " damage" : " healing";
+
+            string damageMessage = $"{absDmg}{action} applied to: ";
+
+            List<string> affectedPlayers = new List<string>();
+
+            foreach (Player targetPlayer in Player.GetAllPlayers())
```

Comment:
PlayerInfo.FindPlayers can be used to search all players.

Then you have to directly use the damage RPC (that is used in the targetPlayer.Damage).

Check https://github.com/JereKuusela/valheim-dev/blob/main/ServerDevcommands/Commands/RPC.cs command for the idea (but I never finished/tested this).

Reply:
I tried to understand how this works, but I was unsuccessful. I think I need to practice reverse engineering this game better. And also my programming skills are pretty bad.
However, you can make any changes to the code if you want to implement it.